### PR TITLE
JDK-8294901: remove pre-VS2017 checks in Windows related coding

### PR DESCRIPTION
**Pull request by MBaesken**
After "8293162: Drop support for VS2017" limited current support to VS2019 and VS2022 it is most likely safe to remove various checks/workarounds related to older VS compilers like VS2015 or VS2013.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8294901](https://bugs.openjdk.org/browse/JDK-8294901): remove pre-VS2017 checks in Windows related coding

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [75d999ec](https://git.openjdk.org/jdk/pull/10600/files/75d999ec5f8f3725d2a220dc10d55b8bfb73982e)
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**)
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10600/head:pull/10600` \
`$ git checkout pull/10600`

Update a local copy of the PR: \
`$ git checkout pull/10600` \
`$ git pull https://git.openjdk.org/jdk pull/10600/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10600`

View PR using the GUI difftool: \
`$ git pr show -t 10600`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10600.diff">https://git.openjdk.org/jdk/pull/10600.diff</a>

</details>

### CODE REVIEW

File: src/java.base/windows/native/libnio/ch/wepoll.c [hunk 1]
```diff
@@ -858,11 +858,6 @@ int err_check_handle(HANDLE handle) {
 
 #define unused_var(v) ((void) (v))
 
-/* Polyfill `inline` for older versions of msvc (up to Visual Studio 2013) */
-#if defined(_MSC_VER) && _MSC_VER < 1900
-#define inline __inline
-#endif
-
```
Comment (AlanBateman) [inline src/java.base/windows/native/libnio/ch/wepoll.c:865]: This is 3rd party code best to leave it out of this change.

Comment (MBaesken) [reply src/java.base/windows/native/libnio/ch/wepoll.c]: I agree, let's leave 3rd party code alone. I adjusted wepoll.c and kept the _MSC_VER check.

**Comment by MBaesken** [comment]
btw while looking at VS-related references,  the build docu seems to be a little outdated too
https://github.com/openjdk/jdk/blob/master/doc/building.md
states  "Windows XP is not a supported platform, but all newer Windows should be able to build the JDK."
That's most likely not true any more , the minimum requirements for VS2017/VS2019 are
https://learn.microsoft.com/en-us/visualstudio/releases/2019/system-requirements
https://learn.microsoft.com/en-us/visualstudio/releases/2017/vs2017-system-requirements-vs
so it is  Windows7 SP1/Windows Server 2016  minimum .
(releases older than that but newer than XP like Vista or Server 2008 are not supported any more)

**Review by TheRealMDoerr — APPROVED** [review]
This version LGTM.

**Review by kimbarrett — APPROVED** [review]
Looks good.

**Pull request closed without merging**